tests/test_deprecations.py

```diff
@@ -115,14 +120,16 @@ def test_missing_target_type():
 def test_deprecated_config():
     """Using the deprecated config format raises a warning."""
     with pytest.warns(UserWarning):
-        Campaign.from_config(deprecated_config)
+        Campaign.from_config(old_style_config)
 
 
 @pytest.mark.parametrize("flag", [False, True])
 def test_deprecated_campaign_tolerance_flag(flag):
     """Constructing a Campaign with the deprecated tolerance flag raises an error."""
     with pytest.raises(DeprecationError):
-        Campaign(None, None, None, numerical_measurements_must_be_within_tolerance=flag)
+        Campaign(
+            Mock(), Mock(), Mock(), numerical_measurements_must_be_within_tolerance=flag
```

Comment:
What does this `Mock()` call here do and why was this changed?

Reply:
the test seemed fine or do you have other reasons for changing it?

Reply:
@AVHopp: there is this general concept of mocking for tests, which I would rather not want  to fully explain here in this thread :D but in a nutshell: creates an object that doesn't complain, no matter what you do with it

@Scienfitz: no, the original `None, None, None` version only worked because we still have missing converters/validators in the `Campaign` class (and others), which will change with the `mypy/hypothesis` PRs. But this PR already added the `to_objective` convenience converter to `Campaign`, which broke the test. Thus the change (which can also stay forever, I think, since the test is only about the keyword)

Reply:
Ok, will resolve then